"npm run build" Fails - error TS2345

> npm --version
> **_6.14.8_**

> node --version
> **_v14.9.0_**

> npx tsc --version
> **_Version 4.1.0-dev.20200829_**

```
> tsjs-lib-generator@ build ..\TSJS-lib-generator-master
> tsc -p ./tsconfig.json && node ./lib/index.js

src/widlprocess.ts:36:89 - error TS2345: Argument of type 'CallbackInterfaceType' is not assignable to parameter of type 'InterfaceType'.
  Property 'inheritance' is missing in type 'CallbackInterfaceType' but required in type 'InterfaceType'.

36             browser["callback-interfaces"]!.interface[rootType.name] = convertInterface(rootType, commentMap);
                                                                                           ~~~~~~~~

  node_modules/@types/webidl2/index.d.ts:157:5
    157     inheritance: string | null;
            ~~~~~~~~~~~
    'inheritance' is declared here.


Found 1 error.

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! tsjs-lib-generator@ build: `tsc -p ./tsconfig.json && node ./lib/index.js`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the tsjs-lib-generator@ build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     %APPDATA%\npm-cache\_logs\2020-08-30T02_04_54_901Z-debug.log
```
_%APPDATA%_\\**npm-cache\_logs\2020-08-30T02_04_54_901Z-debug.log**:
```
0 info it worked if it ends with ok
1 verbose cli [
1 verbose cli   'F:\\Node.js\\nodejs\\node.exe',
1 verbose cli   'F:\\Node.js\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'run',
1 verbose cli   'build'
1 verbose cli ]
2 info using npm@6.14.8
3 info using node@v14.9.0
4 verbose run-script [ 'prebuild', 'build', 'postbuild' ]
5 info lifecycle tsjs-lib-generator@~prebuild: tsjs-lib-generator@
6 info lifecycle tsjs-lib-generator@~build: tsjs-lib-generator@
7 verbose lifecycle tsjs-lib-generator@~build: unsafe-perm in lifecycle true
8 verbose lifecycle tsjs-lib-generator@~build: PATH: F:\Node.js\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin;..\TSJS-lib-generator-master\node_modules\.bin;<%OTHER_PATHS...%>;%APPDATA%\npm
9 verbose lifecycle tsjs-lib-generator@~build: CWD: ..\TSJS-lib-generator-master
10 silly lifecycle tsjs-lib-generator@~build: Args: [ '/d /s /c', 'tsc -p ./tsconfig.json && node ./lib/index.js' ]
11 silly lifecycle tsjs-lib-generator@~build: Returned: code: 2  signal: null
12 info lifecycle tsjs-lib-generator@~build: Failed to exec build script
13 verbose stack Error: tsjs-lib-generator@ build: `tsc -p ./tsconfig.json && node ./lib/index.js`
13 verbose stack Exit status 2
13 verbose stack     at EventEmitter.<anonymous> (F:\Node.js\nodejs\node_modules\npm\node_modules\npm-lifecycle\index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:314:20)
13 verbose stack     at ChildProcess.<anonymous> (F:\Node.js\nodejs\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:314:20)
13 verbose stack     at maybeClose (internal/child_process.js:1047:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)
14 verbose pkgid tsjs-lib-generator@
15 verbose cwd ..\TSJS-lib-generator-master
16 verbose Windows_NT 10.0.18363
17 verbose argv "F:\\Node.js\\nodejs\\node.exe" "F:\\Node.js\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "build"
18 verbose node v14.9.0
19 verbose npm  v6.14.8
20 error code ELIFECYCLE
21 error errno 2
22 error tsjs-lib-generator@ build: `tsc -p ./tsconfig.json && node ./lib/index.js`
22 error Exit status 2
23 error Failed at the tsjs-lib-generator@ build script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 2, true ]
```

The inheritance issue was fixed by @saschanaz... But a new issue is present. Perhaps a `convertCallbackInterface` function should be created, or maybe just accept it in `convertInterface`?

```
src/widlprocess.ts:36:89 - error TS2345: Argument of type 'CallbackInterfaceType' is not assignable to parameter of type 'InterfaceType'.
  Types of property 'type' are incompatible.
    Type '"callback interface"' is not assignable to type '"interface"'.

36             browser["callback-interfaces"]!.interface[rootType.name] = convertInterface(rootType, commentMap);
                                                                                           ~~~~~~~~


Found 1 error.

npm ERR! Test failed.  See above for more details.
```

--- a/package.json
+++ b/package.json
@@ -10,16 +10,16 @@
     "test": "tsc -p ./tsconfig.json && node ./lib/index.js && node ./lib/test.js"
   },
   "dependencies": {
-    "@types/jsdom": "^11.12.0",
-    "@types/node": "^9.6.56",
-    "@types/node-fetch": "^1.6.9",
-    "@types/webidl2": "^23.12.1",
+    "@types/jsdom": "^16.2.4",
+    "@types/node": "^14.6.4",
+    "@types/node-fetch": "^2.5.7",
+    "@types/webidl2": "^23.13.2",
     "cpx2": "^2.0.0",
-    "jsdom": "^11.12.0",
-    "node-fetch": "^2.6.0",
-    "print-diff": "^0.1.1",
+    "jsdom": "^16.4.0",
+    "node-fetch": "^2.6.1",
+    "print-diff": "^1.0.0",
     "styleless-innertext": "^1.1.2",
-    "typescript": "next",
-    "webidl2": "^23.12.1"
+    "typescript": "4.1.0-dev.20200908",
+    "webidl2": "^23.13.0"
   }
 }

--- a/src/widlprocess.ts
+++ b/src/widlprocess.ts
@@ -33,7 +33,7 @@ export function convert(text: string, commentMap: Record<string, string>) {
             browser.namespaces!.push(convertNamespace(rootType, commentMap));
         }
         else if (rootType.type === "callback interface") {
-            browser["callback-interfaces"]!.interface[rootType.name] = convertInterface(rootType, commentMap);
+            browser["callback-interfaces"]!.interface[rootType.name] = convertInterfaceCommon(rootType, commentMap);
         }
         else if (rootType.type === "callback") {
             browser["callback-functions"]!["callback-function"][rootType.name]
@@ -109,7 +109,7 @@ function addComments(obj: any, commentMap: Record<string, string>, container: st
     }
 }
 
-function convertInterfaceCommon(i: webidl2.InterfaceType | webidl2.InterfaceMixinType, commentMap: Record<string, string>) {
+function convertInterfaceCommon(i: webidl2.InterfaceType | webidl2.InterfaceMixinType | webidl2.CallbackInterfaceType, commentMap: Record<string, string>) {
     const result: Browser.Interface = {
         name: i.name,
         extends: "Object",

